source/base/function_lib.cc

```diff
@@ -2483,6 +2483,130 @@ namespace Functions
     return interpolate (data_values, ix, p_unit);
   }
 
+  /* ---------------------- Polynomial ----------------------- */
+
+
+
+  template <int dim>
+  Polynomial<dim>::
+  Polynomial(const Table<2,double> &exponents,
+             const Vector<double>   &coefficients,
+             const unsigned int   n_components)
+    :
+    Function<dim> (n_components),
+    exponents (exponents),
+    coefficients(coefficients)
+  {
+    Assert(exponents.n_rows() == coefficients.size(),
+           ExcDimensionMismatch(exponents.n_rows(), coefficients.size()));
```

Comment:
I think you're trading relatively small amounts of memory for inconsistency. How large are the polynomials you are intending to use? For reasonable meshes, the system matrix can easily have tens or hundreds of MB. I don't assume you will have polynomials that take this much memory.

Besides, if you do have polynomials with this many terms, you probably want to use something like the Horner scheme to evaluate them.